### PR TITLE
Posts - endpoints

### PR DESCRIPTION
**Pull request by jaanonim**
Images for Posts will be implemented leater.

### CODE REVIEW

File: app/routers/posts_router.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+from app import crud
+from app.dependencies import get_db
+from app.models import AccountType, User
+from app.schema import PostResponse, PostCreate, PostEdit
+from app.dependencies import get_current_user
+
+router = APIRouter(tags=["post"])
+
+
+@router.get("/posts", response_model=List[PostResponse])
```
Comment (n2oneProgrammer) [inline app/routers/posts_router.py]: you dont write in docs 404 response

File: app/routers/posts_router.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+):
+    posts = crud.create_post(db, post, current_user)
+    return posts
+
+
+@router.get("/posts/{post_id}", response_model=PostResponse)
+async def get_post(post_id, db: Session = Depends(get_db)):
+    posts = crud.get_post_by_id(db, post_id)
+    if posts is None:
+        raise HTTPException(status_code=404, detail="Post not found")
```
Comment (n2oneProgrammer) [inline app/routers/posts_router.py:49]: you dont write in docs 404 response

File: app/routers/posts_router.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+@router.patch("/posts/{post_id}", response_model=PostResponse)
+async def edite_post(
+    post_id,
+    post_data: PostEdit,
+    db: Session = Depends(get_db),
+    current_user: User = Depends(get_current_user),
+):
+    post = crud.get_post_by_id(db, post_id)
+    if post is None:
+        raise HTTPException(status_code=404, detail="Post not found")
```
Comment (n2oneProgrammer) [inline app/routers/posts_router.py:69]: you dont write in docs 404 response

File: app/routers/posts_router.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    if post.author_id != current_user.id:
+        have_user_permission(current_user, [AccountType.ADMIN])
```
Comment (n2oneProgrammer) [inline app/routers/posts_router.py:71]: you dont write in docs 401response

File: app/routers/posts_router.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+@router.delete("/posts/{post_id}", response_model=PostResponse)
+async def delete_post(
+    post_id,
+    db: Session = Depends(get_db),
+    current_user: User = Depends(get_current_user),
+):
+    posts = crud.get_post_by_id(db, post_id)
+    if posts is None:
+        raise HTTPException(status_code=404, detail="Post not found")
```
Comment (n2oneProgrammer) [inline app/routers/posts_router.py:89]: you dont write in docs 404 response

File: app/schema.py [hunk 6]
```diff
@@ -24,7 +24,31 @@ class Post(BaseModel):
     desc: str
     long: Decimal
     lat: Decimal
-    images: List[Images]
+    # images: List[Images]
```
Comment (n2oneProgrammer) [inline app/schema.py:31]: why class images has images list

File: app/crud.py [hunk 7]
```diff
@@ -23,7 +32,52 @@ def get_user_by_userid(db: Session, user_id: int):
...
+    db.commit()
+    db.refresh(db_item)
+    return db_item
+
+
+def edit_post(db: Session, post: models.Post, post_data: schema.PostEdit):
+    if post_data.title is not None:
+        post.title = post_data.title
+    if post_data.type is not None:
+        post.type = post_data.type
```
Comment (Wuzado) [inline app/crud.py:62]: Should the type be changeable?

File: app/routers/posts_router.py [hunk 8]
```diff
@@ -0,0 +1,67 @@
+from typing import List
+
+from fastapi import Depends, APIRouter, HTTPException
+from sqlalchemy.orm import Session
+
+from app import crud
+from app.dependencies import get_db
+from app.models import AccountType, User
+from app.schema import PostResponse, PostCreate, PostEdit
+from app.dependencies import get_current_user
```
Comment (Wuzado) [inline app/routers/posts_router.py]: I'm not sure, but I think you could shorten `app` to `.`?

File: app/routers/posts_router.py [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+@router.get("/posts/{post_id}", response_model=PostResponse)
+async def get_post(post_id, db: Session = Depends(get_db)):
+    posts = crud.get_post_by_id(db, post_id)
+    if posts is None:
+        raise HTTPException(status_code=404, detail="Post not found")
+    return posts
+
+
+@router.patch("/posts/{post_id}", response_model=PostResponse)
+async def edite_post(
```
Comment (Wuzado) [inline app/routers/posts_router.py]: Typo, should be `edit_post`

File: app/schema.py (hunk 6 above)
Comment (Wuzado) [reply app/schema.py:27]: I think that might be a typo in the class name: should be singular (`Image`) and not plural (`Images`). Other than that, seems ok.

File: app/schema.py [hunk 10]
```diff
@@ -24,7 +24,31 @@ class Post(BaseModel):
...
+        orm_mode = True
+
+
+class PostCreate(BaseModel):
+    title: str
+    type: PostType
+    desc: str
+    long: Decimal
+    lat: Decimal
+    # images: List[Images]
```
Comment (Wuzado) [inline app/schema.py:43]: Wondering: couldn't these Post classes be inherited?

**Review by Wuzado — COMMENTED** [review]
Seems good! Just a couple of small changes that could be useful before merging.

File: app/schema.py (hunk 6 above)
Comment (Wuzado) [reply app/schema.py:31]: I think that might be a typo in the class name: should be singular (Image) and not plural (Images). Other than that, seems ok.

File: app/crud.py (hunk 7 above)
Comment (jaanonim) [reply app/crud.py:62]: Why not?

Comment (Wuzado) [reply app/crud.py:62]: I guess that's something to be addressed in the upcoming meeting. For now, it's ok.

File: app/schema.py (hunk 10 above)
Comment (jaanonim) [reply app/schema.py:43]: because i need to set some fields to be not requested.

Comment (Wuzado) [reply app/schema.py:43]: I see! Closing.

File: app/schema.py (hunk 6 above)
Comment (jaanonim) [reply app/schema.py:31]: thats @n2oneProgrammer mistake

Comment (jaanonim) [reply app/schema.py:31]: I did't wrote that.

File: app/routers/posts_router.py [hunk 11]
```diff
@@ -30,16 +38,27 @@ async def create_post(
...
 
 
-@router.patch("/posts/{post_id}", response_model=PostResponse)
-async def edite_post(
+@router.patch(
+    "/posts/{post_id}",
+    response_model=PostResponse,
+    responses={
+        404: {"description": "Post not found"},
+        401: {"description": "Not authenticated, you must have Admin permission"},
```
Comment (Wuzado) [inline app/routers/posts_router.py:58]: Didn't notice that earlier, but I think users must be allowed to edit their own posts. Is that something in the scope of this PR, or are we moving this change to a later PR?

Comment (jaanonim) [reply app/routers/posts_router.py:58]: This response is send to user if he want to edit someone else post.

Comment (Wuzado) [reply app/routers/posts_router.py:58]: I see!

**Pull request merged**